commons-core/src/test/scala/com/avsystem/commons/serialization/cbor/CborInputOutputTest.scala

```diff
@@ -211,6 +221,19 @@ class CborInputOutputTest extends AnyFunSuite {
   test("chunked byte string") {
     assert(CborInput.readRawCbor[Bytes](RawCbor.fromHex("5F426162426162426162FF")) == Bytes("ababab"))
   }
+
+  test("generic sealed trait") {
+    val success = GenericSealedTrait.Success[Int](234)
+    val successCbor = CborOutput.writeRawCbor[GenericSealedTrait[Int]](success)
+    val decodedSuccess = CborInput.readRawCbor[GenericSealedTrait[Int]](successCbor)
+    assert(success == decodedSuccess)
+
+    val failure = GenericSealedTrait.Failure("error")
+    val failureCbor = CborOutput.writeRawCbor[GenericSealedTrait[String]](failure)
+    val decodedFailure = CborInput.readRawCbor[GenericSealedTrait[String]](failureCbor)
+    assert(failure == decodedFailure)
+  }
```

Comment:
This test does not check if metadata is actually used. If you ignore the metadata completely, this test will still pass.